### PR TITLE
Book an Appointment Capstone - Frontend

### PR DESCRIPTION
**Pull request by jmonto55**
## Description
In this PR we created a UI for users to view a list of houses and make a reservation on individual houses. Also provides a delete house and cancel reservation option.

## Related Issues
[Here👉🏾](https://github.com/jmonto55/book-an-appointment-backend/projects/2) 

## Checklist
In the navigation panel, the user can see links to:

- [x] Reserve form.
- [x]  My reservations.
- [x] Add house.
- [x]  Delete house.
- [x] On the main page, the user can see a list of houses.
- [x] When the user selects a specific item, they can see the [details page]
- [x] In the details page, the user can click the "Reserve" button 
- [x] App is responsive.

## Screenshots
Main Page
image 

Add reservation
image

My reservation
image

Add new House
image

Delete House
image

## Additional Notes
The following took part in developing this work:
1. @mohamed20163858 
2. @jmonto55 
3. @SimonGideon 
4. @DarkxDev 

## ⚠️ Kindly use these credentials to log in ⚠️  
[Click Me to Try it yourself](https://book-an-appointment-frontend-k479bjyzc-jmonto55.vercel.app/)

username: 
```ruby 
mohamed20163858@gmail.com
```
password: 
```ruby
momo123456
```
#### Thanks for your review 🙂

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import HouseReservations from './components/HouseReservations';
+import AddHouse from './components/AddHouse';
+import DeleteHouse from './components/DeleteHouse';
+import ShowHouse from './components/ShowHouse';
+import LoginForm from './components/LoginForm';
+import Splash from './components/Splash';
+import Logout from './components/Logout';
+import SignupForm from './components/SignupForm';
+
+function App() {
```
Comment (Meltrust) [inline src/App.jsx:16]: - [ ]  Please, kindly refactor this and all other functions in your project to ES6 arrow functions, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

You can do a search in Vscode for the "function" word, and correct as needed.  Shouldn't take you more than 5-10 min so don't worry.  

image


That way, your code will meet the latest standards. 💪

File: src/App.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  const { loginToken } = useSelector((store) => store.login);
+  const { logoutToken } = useSelector((store) => store.logout);
+  let isAuthorized = false;
+  if (loginToken || logoutToken) {
+    isAuthorized = true;
+  }
+  console.log('authorized', isAuthorized);
```
Comment (Meltrust) [inline src/App.jsx]: According to the [Javascript best practices general requirements](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md):

image

Therefore:

- [ ] Kindly, remove the debugging code `console.log` present in this file.  

- [ ] Also, kindly remove the debugging code `console.log` present in these files:

image

That way, your repo will be cleaner, and we will meet this particular requirement. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi team,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Nice code organization ✔️ 
-  App is working well✔️ 
-  Good readme ✔️ 
-  Beautiful frontend ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: src/App.jsx (hunk 1 above)
Comment (SimonGideon) [reply src/App.jsx:16]: Noted 👍🏾

**Review by Meltrust — APPROVED** [review]
#### Hi STUDENT_MENTION,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Functions are now ES6 ✔️ 
- Good job✔️ 




Cheers and Happy coding!👏👏👏


------

**Pull request merged**